src/AppInstallerCLICore/ExecutionReporter.h

```diff
@@ -98,13 +100,13 @@ namespace AppInstaller::CLI::Execution
         void SetStyle(AppInstaller::Settings::VisualStyle style);
 
         // Prompts the user, return true if they consented.
-        bool PromptForBoolResponse(Resource::LocString message, Level level = Level::Info);
+        bool PromptForBoolResponse(Resource::LocString message, Level level);

```

Comment:
We forgot the PromptForBoolResponse in ConfigurationFlow.cpp, which also needs to be handled. And that one is almost always prompted. It's the "Do you want to apply the device configuration?" before winget configure apply changes to the device.

Check IsEnabled() within PromptFor* has one benefit that all future invocations do not need to remember to check reporter level before prompt. Regarding the concern that different caller expect different results, we can make this method like `bool PromptForBoolResponse(Resource::LocString message, Level level, bool resultIfDisabled)` so in case the level is disabled, just return resultIfDisabled.

Same for PromptForPath, where caller can say return empty string is disabled. Then treat empty string as invalid imput and return "Install Location not provided."

But I'm ok with either way at the moment.

Reply:
Makes complete sense to me - Since checking within the PromptFor* means one less thing to remember, I went with that.